system/memctrl.h

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: GPL-2.0
+#ifndef MEMCTRL_H
+#define MEMCTRL_H
+/**
+ * \file
+ *
+ * Provides information about the memory controller status
+ * (running DRAM configuration, ECC, ...) and other
+ * platform-specific data
+ *
+ *//*
+ * Copyright (C) 2004-2023 Sam Demeulemeester.
+ */
+
+typedef struct imc_infos {
+    uint16_t    freq;
+    char        *type;
+    uint16_t    width;
+    uint16_t    tCL;
+    uint8_t     tCL_dec;
+    uint16_t    tRCD;
+    uint16_t    tRP;
+    uint16_t    tRAS;
+} imc_info_t;
+
+typedef enum {
+    ECC_ERR_NONE,
+    ECC_ERR_CORRECTED,
+    ECC_ERR_UNCORRECTED
+} ecc_error_typo_t;
+
+typedef struct ecc_status {
+    bool                ecc_enabled;
+    ecc_error_typo_t    err_type;
+    uint64_t            err_adr;
```

Comment:
More padding happens in this struct, to align err_type on a 32-bit boundary and err_addr on a 64-bit boundary.